spec/lucky/subdomain_spec.cr

```diff
@@ -0,0 +1,129 @@
+require "../spec_helper"
+
+include ContextHelper
+
+abstract class BaseAction < Lucky::Action
+  include Lucky::Subdomain
+  accepted_formats [:html], default: :html
+end
+
+class Simple::Index < BaseAction
+  register_subdomain
+
+  get "/simple" do
+    plain_text subdomain
+  end
+end
+
+class Specific::Index < BaseAction
+  register_subdomain "foo"
```

Comment:
Does this work for multiple subdomain too? Like `"staging.dashboard.foo"`

Reply:
My current job uses "multiple subdomains" as well because we have a tenant subdomain and the environment subdomain so in staging a url might look like `tenant1.staging.example.com`.

This subdomain support really focuses on 1 subdomain. Without any special handling, the subdomain for my example would be `tenant1.staging` and it would be up to me to add special logic to parse it further.

Another way to handle that would be to configure the `tld_length` to be 2 only in staging so that the subdomain would become `tenant1` but I think that might be a sort of hack.

Reply:
Ok, yeah, that's what my sites do too. `staging.tenant1.example.com`. That's fine though. We just would document that on the site saying if you use deep nested subdomains, then you'll need to do your own split and parse.